Source/Models/Highlight.swift

```diff
@@ -7,139 +7,129 @@
 //
 
 import Foundation
+import RealmSwift
 import CoreData
 
 public typealias Completion = (error: NSError?) -> ()
-let coreDataManager = CoreDataManager()
 
-@objc(Highlight)
-public class Highlight: NSManagedObject {
+class Highlight: Object {
     
-    public func persist(completion: Completion?) {
-        var highlight: Highlight?
-        
-        do {
-            let fetchRequest = NSFetchRequest(entityName: "Highlight")
-            fetchRequest.predicate = NSPredicate(format:"highlightId = %@", highlightId)
-            highlight = try coreDataManager.managedObjectContext.executeFetchRequest(fetchRequest).last as? Highlight
-        } catch let error as NSError {
-            print(error)
-            highlight = nil
-        }
-        
-        if highlight != nil {
-            highlight!.content = content
-            highlight!.contentPre = contentPre
-            highlight!.contentPost = contentPost
-            highlight!.date = date
-            highlight!.type = type
-        } else {
-            highlight = NSEntityDescription.insertNewObjectForEntityForName("Highlight", inManagedObjectContext: coreDataManager.managedObjectContext) as? Highlight
-            
-            highlight!.bookId = bookId
-            highlight!.content = content
-            highlight!.contentPre = contentPre
-            highlight!.contentPost = contentPost
-            highlight!.date = date
-            highlight!.highlightId = highlightId
-            highlight!.page = page
-            highlight!.type = type
-        }
-        
-        // Save
-        do {
-            try coreDataManager.managedObjectContext.save()
-            completion?(error: nil)
-        } catch let error as NSError {
-            completion?(error: error)
-        }
+    dynamic var bookId:String!
+    dynamic var content:String!
+    dynamic var contentPost:String!
+    dynamic var contentPre:String!
+    dynamic var date:NSDate!
+    dynamic var highlightId:String!
+    dynamic var page:Int = 0
+    dynamic var type:Int = 0
+    dynamic var startOffset:Int = -1
+    dynamic var endOffset:Int = -1
+    
+    override class func primaryKey()-> String{
+        return "highlightId"
     }
     
     public static func persistHighlight(object: FRHighlight, completion: Completion?) {
```

Comment:
Something like this:
`public static func persistHighlight(object: FRHighlight, completion: Completion? = nil) {`